CATROID-404 Insert existing user list as parameter of list functions

mstoeg opened the pull request:
- Refactor/Move tests out of FormulaEditorFunctionListTest
- Add UI espresso test to check if the last user list is inserted correctly.
- Add Unit test to check if Dialog pops up when there is no current user list.

https://jira.catrob.at/browse/CATROID-404

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

gPathpp commented on catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CategoryListFragment.java:
```diff
@@ -238,47 +243,115 @@ public void onItemClick(CategoryListItem item) {
+					RadioButton addToProjectListsRadioButton = ((Dialog) dialog).findViewById(R.id.global);
+					boolean addToProjectLists = addToProjectListsRadioButton.isChecked();
+
+					if (addToProjectLists) {
```
Please make a util method/class, or something else for this task. Because there are already two places where the user-list is added to the project.

mstoeg commented on catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CategoryListFragment.java:
```diff
@@ -238,47 +243,115 @@ public void onItemClick(CategoryListItem item) {
+		FragmentActivity activity = getActivity();
+		TextInputDialog.Builder builder = new TextInputDialog.Builder(activity);
+		final List<UserList> projectUserList =
+				ProjectManager.getInstance().getCurrentProject().getUserLists();
```
According to @SimonStefan this is not yet possible for fragments. As it will still take time till the concerning PR is merged ProjectManager.getInstance() should be used until then.

gPathpp replied:
Ok, thanks!

gPathpp commented on catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternFormula.java:
```diff
@@ -726,6 +726,11 @@ private CursorTokenPropertiesAfterModification setCursorPositionAndSelectionAfte
 						.getFunctionParameterInternTokensAsLists(functionInternTokenList);
 				List<InternToken> functionFirstParameter = functionParameters.get(0);
 
+				if (functionInternTokenList.get(0).getTokenStringValue().equals(Functions.LIST_ITEM.name()) && !functionFirstParameter.get(0).isUserList()) {
```
Please add a test.

gPathpp commented on catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/CategoryListRVAdapter.java:
```diff
@@ -89,12 +89,7 @@ public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
 		}
 
 		holder.title.setText(item.text);
-		holder.itemView.setOnClickListener(new View.OnClickListener() {
```
Nice!

gPathpp commented on catroid/src/main/java/org/catrobat/catroid/utils/AddUserListDialog.java:
```diff
@@ -0,0 +1,89 @@
+	}
+
+	public void show(Callback callback) {
+		builder.setHint(activity.getString(R.string.data_label))
```
Please remove the activity and pass the strings instead.

mstoeg commented:
@wslany - thank you for the response! I have overlooked that a new User List Function was implemented in the meantime. I fixed the issue and also added a testcase for the correct order of the new function "item's index".

wslany commented:
Thanks!
